ambry-frontend/src/main/java/com.github.ambry.frontend/AmbryBlobStorageService.java

```diff
@@ -418,8 +445,8 @@ public void onCompletion(String result, Exception exception) {
         }
       }
 
-      if (exception != null) {
-        submitResponse(restRequest, restResponseChannel, null, exception);
+      if (response != null || exception != null) {
+        submitResponse(restRequest, restResponseChannel, response, exception);
```

Comment:
is there a place where we decide between sending a response or an exception if both are provided for some reason?